File: Exemplos/App_IntentDigitalHUB_Kotlin/Elgin_IntentDigitalHUB_Java-font/app/src/main/java/com/elgin/intent_digital_hub/IntentDigitalHubService/SAT/SatCommand.kt
package com.elgin.intent_digital_hub.IntentDigitalHubService.SAT

import com.elgin.intent_digital_hub.IntentDigitalHubService.IntentDigitalHubCommand
import com.elgin.intent_digital_hub.IntentDigitalHubService.IntentDigitalHubModule
import com.google.gson.annotations.SerializedName
import java.io.Serializable

/**
 * Classe que generaliza todos os comandos SAT definindo o módulo do comando e o seu tipo de retorno
 */
abstract class SatCommand protected constructor(functionName: String?) :
    IntentDigitalHubCommand(functionName, IntentDigitalHubModule.SAT), Serializable {
    //O retorno dos comandos SAT é sempre uma String
    @SerializedName("resultado")
    val resultado: String? = null
}

File: Exemplos/App_IntentDigitalHUB_Ionic/Elgin_IntentDigitalHUB_Ionic-font/src/intentDigitalHubService/SAT/SatCommand.ts
import { IntentDigitalHubCommand } from "../IntentDigitalHubCommand";
import { IntentDigitalHubModule } from "../IntentDigitalHubModule";

/**
 * Classe que generaliza todos os comandos SAT definindo o módulo do comando e o seu tipo de retorno
 */
export abstract class SatCommand extends IntentDigitalHubCommand {
  constructor(functionName: string) {
    super(functionName, IntentDigitalHubModule.SAT);
  }
}
